lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -657,8 +657,12 @@ sub overview {
         groups => $groups,
         until => $until,
     );
-    my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
-    ($stash{archs}, $stash{results}, $stash{aggregated}) = $self->prepare_job_results(\@latest_jobs);
+    my $limit = $search_args->{limit} = OpenQA::App->singleton->config->{misc_limits}->{tests_overview_max_jobs};
+    my $jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args);
+    my $limit_exceeded = $jobs->count >= $limit ? $limit : undef;
+    my @latest_jobs = $jobs->latest_jobs($until);
```

Comment:
Since `latest_jobs` might narrow down the number of jobs further it would make sense to evaluate whether the limit is exceeded only after that:

```suggestion
    my @latest_jobs = $jobs->latest_jobs($until);
    my $limit_exceeded = scalar @latest_jobs >= $limit ? $limit : undef;
```

Reply:
Just moving the check doesn't work for the case the ticket is about, i'll have to revisit the whole logic.

Reply:
See other comment for explanation: https://github.com/os-autoinst/openQA/pull/4672#discussion_r880727332